src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/__init__.py

```diff
@@ -4,19 +4,24 @@
 # --------------------------------------------------------------------------------------------
 
 from azure.cli.core import AzCommandsLoader
-from azure.cli.core.profiles import ResourceType
+from azure.cli.core.profiles import ResourceType, CustomResourceType
 from azure.cli.core.commands import AzCommandGroup, AzArgumentContext
 
 import azure.cli.command_modules.storage._help  # pylint: disable=unused-import
 
 
+CUSTOM_MGMT_STORAGE = CustomResourceType('azure.mgmt.storage', 'StorageManagementClient')
+
+
 class StorageCommandsLoader(AzCommandsLoader):
     def __init__(self, cli_ctx=None):
         from azure.cli.core.commands import CliCommandType
 
         storage_custom = CliCommandType(operations_tmpl='azure.cli.command_modules.storage.custom#{}')
+        custom_resource_types = {'latest': {CUSTOM_MGMT_STORAGE: '2015-06-15'}}
         super(StorageCommandsLoader, self).__init__(cli_ctx=cli_ctx,
                                                     resource_type=ResourceType.DATA_STORAGE,
+                                                    custom_resource_types=custom_resource_types,
```

Comment:
Kind of strange adding it here. The other kwargs are more sort of defaults that the commands from this loader will use.
The `custom_resource_types` kwargs is used to inject new resourcetypes to the CLI. 
Calling a function from core might make more sense.

Reply:
Moved to calling a function.